pkg/services/ngalert/notifier/receiver_svc_test.go

```diff
@@ -0,0 +1,208 @@
+package notifier
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/infra/db"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/accesscontrol/acimpl"
+	"github.com/grafana/grafana/pkg/services/accesscontrol/actest"
+	"github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/tests/fakes"
+	"github.com/grafana/grafana/pkg/services/secrets"
+	"github.com/grafana/grafana/pkg/services/secrets/database"
+	"github.com/grafana/grafana/pkg/services/secrets/manager"
+	"github.com/grafana/grafana/pkg/services/user"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReceiverService_GetReceivers(t *testing.T) {
+	sqlStore := db.InitTestDB(t)
+	secretsService := manager.SetupTestService(t, database.ProvideSecretsStore(sqlStore))
+
+	t.Run("service gets receiver groups from AM config", func(t *testing.T) {
+		sut := createReceiverServiceSut(t, secretsService)
+
+		Receivers, err := sut.GetReceivers(context.Background(), rQuery(1), nil)
+		require.NoError(t, err)
+		require.Len(t, Receivers, 2)
+		require.Equal(t, "grafana-default-email", Receivers[0].Name)
+		require.Equal(t, "slack receiver", Receivers[1].Name)
+	})
+
+	t.Run("service filters receiver groups by name", func(t *testing.T) {
+		sut := createReceiverServiceSut(t, secretsService)
+
+		Receivers, err := sut.GetReceivers(context.Background(), rQuery(1, "slack receiver"), nil)
+		require.NoError(t, err)
+		require.Len(t, Receivers, 1)
+		require.Equal(t, "slack receiver", Receivers[0].Name)
+	})
+}
```

Comment:
Tests for pagination would be great but can be added later, I think, when API is implemented